packages/smarthr-ui/src/components/Table/ThCheckbox.tsx

```diff
@@ -1,50 +1,63 @@
 import React, { forwardRef, useMemo } from 'react'
 import { tv } from 'tailwind-variants'
 
+import { Balloon } from '../Balloon'
 import { CheckBox, Props as CheckBoxProps } from '../CheckBox'
 import { Center } from '../Layout'
-import { VisuallyHiddenText } from '../VisuallyHiddenText'
 
 import { Th } from './Th'
 
 import type { DecoratorsType } from '../../types'
 
 type Props = {
-  decorators?: DecoratorsType<'checkAllInvisibleLabel'>
+  decorators?: DecoratorsType<'checkAllInvisibleLabel'> & {
+    checkColumnName?: (text: string) => string
+  }
 }
 
-const CHECK_ALL_INVISIBLE_LABEL = 'すべての行を選択'
+const CHECK_ALL_INVISIBLE_LABEL = 'すべての項目を選択/解除'
+const CHECK_COLUMN_NAME = '選択'
 
 const thCheckbox = tv({
   slots: {
-    inner: 'shr-absolute shr-inset-0 [&:not(:has([disabled]))]:shr-cursor-pointer',
+    inner: 'shr-absolute shr-inset-0 [&:not(:has([disabled]))]:shr-cursor-pointer shr-group/label',
     wrapper: 'shr-relative shr-w-[theme(fontSize.base)] [&]:shr-px-0.75',
     checkbox: '[&]:shr-block',
+    balloon: [
+      // 位置はセルの真ん中(50%)+checkboxの幅の半分(8px)+outlineの幅(4px)+Balloonの矢印の幅(5px), sr-onlyで隠す
+      'shr-absolute shr-left-[calc(50%+(theme(fontSize.base)/2)+4px+5px)] shr-sr-only',
+      // labelの中の要素に hover or focus-visible がある時のスタイル。shr-absoluteはshr-not-sr-onlyのpositionをabsoluteに上書きしている
+      'group-has-[:hover,:focus-visible]/label:shr-not-sr-only group-has-[:hover,:focus-visible]/label:shr-absolute group-has-[:hover,:focus-visible]/label:shr-whitespace-nowrap',
```

Comment:
`group-has-[]` が Firefox で機能していませんでした。`group-[:has()]` で回避はできそうです。

```suggestion
      'group-[:has(:hover,:focus-visible)]/label:shr-not-sr-only group-[:has(:hover,:focus-visible)]/label:shr-absolute group-[:has(:hover,:focus-visible)]/label:shr-whitespace-nowrap',
```

Reply:
@uknmr 今の記載の方法でもFirefoxでも動いてると思うんだけど、ちょっと気づいたのは他のStoryからTableのStoryに遷移してきたタイミングだと、機能してないかも :sob: （再読み込みすると動く）

Reply:
謎ですね。Storybook 環境起因の可能性ありそうなので、棄却してプロダクトで問題出たら別途起票します!